### PR TITLE
Add docker stream adapter to read streams from docker

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

**Comment by cjh1** [comment]
@zachmullen Here is the test cases. The output of the testDockerModeStdErrStdOut test case is:

```
FAILED (failures=2, errors=1)
39: {'_stderr': {'data': '\x02\x00\x00\x00\x00\x00\x00\x14this is stderr data\n', 'format': 'string'}, '_stdout': {'data': '\x01\x00\x00\x00\x00\x00\x00\x14this is stdout data\n', 'format': 'string'}}
```

You can see the header information that need to stripped off. There header format is described here:

https://docs.docker.com/engine/api/v1.24/, search for 'ATTACH TO A CONTAINER' ( subheading 'Stream details' )

**Comment by cjh1** [comment]
@zachmullen Looks like this might be relevant: https://github.com/docker/docker-py/pull/900/files, shame it not merged.

**Comment by cjh1** [comment]
@zachmullen Here you go, PTAL, this should fix things up.

File: girder_worker/plugins/docker/executor.py [hunk 1]
```diff
@@ -197,10 +198,10 @@ def exit_condition():
         def close_output(output):
             return output not in (stdout.fileno(), stderr.fileno())
 
-        opipes[stdout.fileno()] = opipes.get(
-            '_stdout', utils.WritePipeAdapter({}, sys.stdout))
-        opipes[stderr.fileno()] = opipes.get(
-            '_stderr', utils.WritePipeAdapter({}, sys.stderr))
+        opipes[stdout.fileno()] = DockerStreamPushAdapter(opipes.get(
```
Comment (zachmullen) [inline girder_worker/plugins/docker/executor.py]: Heh, this is exactly the approach I was imagining yesterday 👍

**Comment by cjh1** [comment]
@zachmullen Just fixed a typo, no code changes

**Comment by zachmullen** [comment]
Merging this so I can rebase my topic

**Pull request merged**